From: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
Subject: [PATCH v4 4/4] wdscore: Add stubs for WdsSetupLogMessage[AW].
Message-Id: <20220127232102.181453-4-mohamadaljaf@gmail.com>
Date: Thu, 27 Jan 2022 18:21:02 -0500
In-Reply-To: <20220127232102.181453-1-mohamadaljaf@gmail.com>
References: <20220127232102.181453-1-mohamadaljaf@gmail.com>

Required by the Windows MediaCreationTool21H2.

Signed-off-by: Mohamad Al-Jaf <mohamadaljaf@gmail.com>
---
v4: - Add missing argument
    - Rename one more unknown
---
 dlls/wdscore/main.c             | 28 ++++++++++++++++++++++++++++
 dlls/wdscore/wdscore.spec       |  4 ++--
 dlls/wdscore/wdscore_internal.h | 27 +++++++++++++++++++++++++++
 3 files changed, 57 insertions(+), 2 deletions(-)

diff --git a/dlls/wdscore/main.c b/dlls/wdscore/main.c
index 6f054cdd63b..e4d28b9301a 100644
--- a/dlls/wdscore/main.c
+++ b/dlls/wdscore/main.c
@@ -74,3 +74,31 @@ LPVOID WINAPI ConstructPartialMsgVW( WdsLogLevel level, LPCWSTR msg, va_list arg
     FIXME( "%u %s - stub\n", level, debugstr_w(msg) );
     return NULL;
 }
+
+
+/***********************************************************************
+ *           WdsSetupLogMessageA (wdscore.@)
+ */
+HRESULT WINAPI WdsSetupLogMessageA( LPVOID msg, WdsLogSource src, LPCSTR unknown1, LPCSTR unknown2,
+                                    ULONG unknown3, LPCSTR file, LPCSTR func, void *ip, 
+                                    ULONG unknown4, void *unknown5, UINT unknown6 )
+{
+    FIXME( "%p, %u, %s, %s, %u, %s, %s, %p, %u, %p, %u - stub\n", msg, src, debugstr_a(unknown1),
+           debugstr_a(unknown2), unknown3, debugstr_a(file), debugstr_a(func), ip, unknown4, 
+           unknown5, unknown6 );
+    return S_OK;
+}
+
+
+/***********************************************************************
+ *           WdsSetupLogMessageW (wdscore.@)
+ */
+HRESULT WINAPI WdsSetupLogMessageW( LPVOID msg, WdsLogSource src, LPCWSTR unknown1, LPCWSTR unknown2,
+                                    ULONG unknown3, LPCWSTR file, LPCWSTR func, void *ip, 
+                                    ULONG unknown4, void *unknown5, UINT unknown6 )
+{
+    FIXME( "%p, %u, %s, %s, %u, %s, %s, %p, %u, %p, %u - stub\n", msg, src, debugstr_w(unknown1),
+           debugstr_w(unknown2), unknown3, debugstr_w(file), debugstr_w(func), ip, unknown4, 
+           unknown5, unknown6 );
+    return S_OK;
+}
diff --git a/dlls/wdscore/wdscore.spec b/dlls/wdscore/wdscore.spec
index b14778001a8..7ec8025ddee 100644
--- a/dlls/wdscore/wdscore.spec
+++ b/dlls/wdscore/wdscore.spec
@@ -150,8 +150,8 @@
 @ stub WdsSetUILanguage
 @ stub WdsSetupLogDestroy
 @ stub WdsSetupLogInit
-@ stub WdsSetupLogMessageA
-@ stub WdsSetupLogMessageW
+@ stdcall WdsSetupLogMessageA(ptr long str str long str str ptr long ptr long)
+@ stdcall WdsSetupLogMessageW(ptr long wstr wstr long wstr wstr ptr long ptr long)
 @ stub WdsSubscribeEx
 @ stub WdsTerminate
 @ stub WdsUnlockExecutionGroup
diff --git a/dlls/wdscore/wdscore_internal.h b/dlls/wdscore/wdscore_internal.h
index b5cbe3c8c25..84a17d5fbc0 100644
--- a/dlls/wdscore/wdscore_internal.h
+++ b/dlls/wdscore/wdscore_internal.h
@@ -30,4 +30,31 @@ typedef enum _WdsLogLevel {
     WdsLogLevelTrace      = 0x7000000
 } WdsLogLevel;
 
+typedef enum _WdsLogSource {
+    WdsLogSourceDPX    = 0x1000000,
+    WdsLogSourceCBS    = 0x2000000,
+    WdsLogSourceCSI    = 0x1800000,
+    WdsLogSourceSXS    = 0x2800000,
+    WdsLogSourceCMI    = 0x3000000,
+    WdsLogSourceDEPLOY = 0x4000000,
+    WdsLogSourceDU     = 0x5000000,
+    WdsLogSourceIBS    = 0x6000000,
+    WdsLogSourceIBSLIB = 0x6400000,
+    WdsLogSourceDIAG   = 0x7000000,
+    WdsLogSourceDIAGER = 0x7400000,
+    WdsLogSourceMIG    = 0x8000000,
+    WdsLogSourceHWARE  = 0x8400000,
+    WdsLogSourceMIGUI  = 0x8800000,
+    WdsLogSourceUI     = 0xA000000,
+    WdsLogSourceCONX   = 0xA400000,
+    WdsLogSourceMOUPG  = 0xA800000,
+    WdsLogSourceWDS    = 0xB000000,
+    WdsLogSourceDISM   = 0xB800000,
+    WdsLogSourcePANTHR = 0x9000000,
+    WdsLogSourceWINPE  = 0xC000000,
+    WdsLogSourceSP     = 0xC800000,
+    WdsLogSourceLIB    = 0xD000000,
+    WdsLogSourceTOOL   = 0xE000000
+} WdsLogSource;
+
 #endif /* __WDSCORE_INTERNAL_H */

-- 
2.35.1

